drivers/include/net/netdev2.h

```diff
@@ -20,13 +20,19 @@
  *
  * Example call flow:
  *
- * 1. packet arrives -> driver calls netdev2->event_callback with
- *    event==NETDEV_EVENT_ISR (from interrupt routine)
- * 2. event_callback wakes up upper layer thread
- * 3. upper layer calls netdev2->driver.isr()
- * 4. netdev2->driver.isr() calls netdev2->event_callback() with
- *    event==NETDEV_EVENT_RX_COMPLETE
- * 5. netdev2->event_callback() uses netdev2->driver.recv() to fetch packet
+ * 1. packet arrives for device
+ * 2. driver calls @ref netdev2_t::event_callback "netdev->event_callback()"
```

Comment:
In general I would refer to what I remember about basic principles of documentation.

  - *Try to avoid information that is not essential for understanding the specific topic.*
  - *Use a figure only to provide additional structural overview. Don't repeat information which is already in your text. Instead try to use references to the explaining passages.*  
  Have you considered just using the sequence numbers as message labels in the SD? That is space saving too.
  - *Always remember that the reader is someone who needs your guide through the topic. And not someone who is capable to resolve inconsistencies on its own.*  
   You claimed the different notations to be all correct. Indeed it is true.  
   But if **I** see different notaions, slightly different function names, etc. Then I try to figure out if the is some meaning in those different representations or I check if I'm the wrong part of the diagram. For that moment I'm completely distracted from the sequence of events you want to show me.

Phew, that turned into something like the "Wort zum Sonntag" :D
Not intended to offend you. I just want to make clear why I'm stressing those *stupid questions* so much.

>> driver calls netdev->event_callback() with `event` := NETDEV2_EVENT_ISR ***(from Interrupt Service Routine)*** and wakes up event handler

> Any idea how to make it more clear?

Just try what you answered to my questions to get me on the track.

Before the start of the sequence, mention that the ISR has to be registered by the driver as a precondition.
And then e.g.:
1. ISR calls netdev->event_callback(NETDEV2_EVENT_ISR) and wakes up event handler

Given that I understood what is going on...